go-controller/pkg/util/node_annotations.go

```diff
@@ -464,6 +504,36 @@ func NodeTransitSwitchPortAddrAnnotationChanged(oldNode, newNode *corev1.Node) b
 	return oldNode.Annotations[ovnTransitSwitchPortAddr] != newNode.Annotations[ovnTransitSwitchPortAddr]
 }
 
+func NodeZoneClusterChanged(oldNode, newNode *corev1.Node, networkName string, newNodeIsLocalZone bool) bool {
+	// Check if the annotations have changed. Use network topology and local params to skip unecessary checks
+
+	isDefaultNetwork := networkName == types.DefaultNetworkName
+	isSecondaryNetwork := !isDefaultNetwork
```

Comment:
you can pass in `NetInfo` where you have method `IsSecondary` which is how we are checking this out generally throughout the code.